lib/mongodb/native_mongodb.nit

```diff
@@ -0,0 +1,501 @@
+	fun remove_all(selector: NativeBSON): Bool import set_mongoc_error `{
+		bson_error_t error;
+		if(!mongoc_collection_remove(recv, MONGOC_REMOVE_NONE, selector, NULL, &error)) {
+			NativeMongoCollection_set_mongoc_error(recv, &error);
+			return false;
+		}
+		return true;
+	`}
+
+	# Wrapper for `mongoc_collection_update(MONGOC_UPDATE_NONE)`.
+	#
+	# This function shall update the first document in the collection that
+	# matches `selector`.
+	fun update(selector, update: NativeBSON): Bool import set_mongoc_error `{
+		bson_error_t error;
+		if(!mongoc_collection_update(recv, MONGOC_UPDATE_NONE, selector, update, NULL, &error)) {
+			NativeMongoCollection_set_mongoc_error(recv, &error);
+			return false;
+		}
+		return true;
+	`}
+
+	# Wrapper for `mongoc_collection_update(MONGOC_UPDATE_MULTI_UPDATE)`.
+	#
+	# This function shall update documents in the collection that match `selector`.
+	fun update_all(selector, update: NativeBSON): Bool import set_mongoc_error `{
+		bson_error_t error;
+		if(!mongoc_collection_update(recv, MONGOC_UPDATE_MULTI_UPDATE, selector, update, NULL, &error)) {
+			NativeMongoCollection_set_mongoc_error(recv, &error);
+			return false;
+		}
+		return true;
+	`}
+
+	# Wrapper for `mongoc_collection_count()`.
+	#
+	# This function shall execute a count `query` on the underlying collection.
+	fun count(query: NativeBSON): Int import set_mongoc_error `{
+		bson_error_t error;
+		int64_t count = mongoc_collection_count(recv, MONGOC_QUERY_NONE, query, 0, 0, NULL, &error);
+		if(count < 0) {
+			NativeMongoCollection_set_mongoc_error(recv, &error);
+			return -1;
+		}
+		return count;
+	`}
+
+	# Wrapper for `mongoc_collection_find()`.
+	#
+	# This function shall execute a `query` on the underlying collection.
+	#
+	# If no options are necessary, `query` can simply contain a query such as `{a:1}`.
+	#
+	# If you would like to specify options such as a sort order,
+	# the query must be placed inside of `{"$query": {}}`.
+	fun find(query: NativeBSON): nullable NativeMongoCursor import
+		NativeMongoCursor.as nullable, set_mongoc_error `{
+		bson_error_t error;
+		mongoc_cursor_t	*cursor;
+		cursor = mongoc_collection_find(recv, MONGOC_QUERY_NONE, 0, 0, 0, query, NULL, NULL);
+
+		if (mongoc_cursor_error(cursor, &error)) {
+			NativeMongoCollection_set_mongoc_error(recv, &error);
+			return null_NativeMongoCursor();
+		}
+
+		return NativeMongoCursor_as_nullable(cursor);
+	`}
+
+	# Wrapper for `mongoc_collection_stats()`.
+	#
+	# This function is a helper to retrieve statistics about the collection.
+	fun stats: nullable NativeBSON import set_mongoc_error, NativeBSON.as nullable `{
+		bson_error_t error;
+		bson_t reply;
+		if(!mongoc_collection_stats(recv, NULL, &reply, &error)){
+			NativeMongoCollection_set_mongoc_error(recv, &error);
+			return null_NativeBSON();
+		}
+		//FIXME how to free the reply allocated by mongoc?
+		return NativeBSON_as_nullable(&reply);
```

Comment:
The `&` here is a problem, you return a reference to a variable on the stack. You need to malloc `replay`, and free it later.